src/ui/src/core_components/base/BaseInputRange.vue

```diff
@@ -0,0 +1,212 @@
+<template>
+	<div class="BaseInputRange">
+		<transition>
+			<div
+				v-show="isPopoverDisplayed"
+				class="BaseInputRange__popover"
+				:style="{ left: popoverLeft }"
+			>
+				<span class="BaseInputRange__popover__content">
+					{{ value }}
+				</span>
+			</div>
+		</transition>
+		<div ref="slider" class="BaseInputRange__slider">
+			<div
+				class="BaseInputRange__slider__progress"
+				:style="{ width: `calc(${progress}% + ${thumbRadius}px)` }"
+			></div>
+		</div>
+		<button
+			ref="thumb"
+			type="button"
+			class="BaseInputRange__thumb"
+			:style="{ left: thumbLeft }"
+			@keydown.left="updateValue(+model - 1)"
+			@keydown.right="updateValue(+model + 1)"
+			@mousedown="handleMouseDown"
+		></button>
+	</div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from "vue";
+
+const props = defineProps({
+	min: { type: Number, required: false, default: 0 },
+	max: { type: Number, required: false, default: 100 },
+	step: { type: Number, required: false, default: undefined },
+});
+
+const model = defineModel("value", { type: Number, default: 50 });
+
+const thumbRadius = 9;
+
+const thumb = ref<HTMLElement>();
+const slider = ref<HTMLElement>();
+
+const progress = computed(
+	() => ((model.value - props.min) / (props.max - props.min)) * 100,
+);
+
+const isPopoverDisplayed = ref(false);
+let popoverTimeout = null;
+
+function displayPopover() {
+	isPopoverDisplayed.value = true;
+
+	if (popoverTimeout) {
+		clearTimeout(popoverTimeout);
+		popoverTimeout = null;
+	}
+
+	popoverTimeout = setTimeout(() => {
+		isPopoverDisplayed.value = false;
+		popoverTimeout = null;
+	}, 1_000);
+}
+
+// clamp(0px, calc(62% - 9px), calc(100% - 18px))
+const thumbLeft = computed(
+	() => `clamp(
+		0px,
+		calc(${progress.value}% - ${thumbRadius}px),
+		calc(100% - ${thumbRadius * 2}px)
+	)`,
+);
+
+const popoverLeft = computed(() => `calc(${thumbLeft.value} - 3px)`);
+
+function updateValue(value: number) {
+	displayPopover();
+	if (props.min !== undefined && value < props.min) return;
+	if (props.max !== undefined && value > props.max) return;
+	model.value = value;
+}
+
+function handleMouseDown() {
+	document.addEventListener("mousemove", onMouseMove);
+	document.addEventListener("mouseup", onMouseUp);
+
+	const sliderBoundingRect = slider.value.getBoundingClientRect();
+
+	function onMouseMove(event: MouseEvent) {
+		const progress =
+			(event.x - sliderBoundingRect.left) /
+			(sliderBoundingRect.right - sliderBoundingRect.left);
+
+		if (progress > 1 || progress < 0) return;
+
+		const value = Math.round(
+			(props.max - props.min) * progress + props.min,
+		);
```

Comment:
We might not want to round since `min/max` can be floating. We might want to round using `props.step`